File: public/src/Model/point.ts
class Point {
    x: number;
    y: number;

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }

    equals(other: Point): boolean {
        if (this.x === other.x && this.y === other.y) return true;

        return false;
    }
}

export default Point;

File: addons/algs/src/point.cpp
#include "../point.h";

Point::Point()
{
	x = -1;
	y = -1;
}

Point::Point(int x, int y)
{
	this->x = x;
	this->y = y;
}

bool Point::operator==(const Point& other) const
{
	return x == other.x && y == other.y;
}

bool Point::operator!=(const Point& other) const
{
	return x != other.x || y != other.y;
}